src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+
+More discussion and examples illustrating how sample size and normalization methods can affect alpha diversity metrics in references 4 and 5.
+
+In this example, we're going to calculate the diversity measures for our data with no normalization, our CSS normalized data, and our rarefied data.
+
+1. First, let's define a function for calculating the diversity measures for a given assay.
+
+```{r}
+calculate_diversity_measures <- function(tse, assay_type) {
+  colData(tse)[[paste0("observed_", assay_type)]] <- estimateRichness(tse, index = "observed", assay.type = assay_type)$observed
+  colData(tse)[[paste0("chao1_", assay_type)]] <- estimateRichness(tse, index = "chao1", assay.type = assay_type)$chao1
+  colData(tse)[[paste0("ace_", assay_type)]] <- estimateRichness(tse, index = "ace", assay.type = assay_type)$ace
+  colData(tse)[[paste0("shannon_", assay_type)]] <- estimateDiversity(tse, index = "shannon", assay.type = assay_type)$shannon
+  colData(tse)[[paste0("simpson_", assay_type)]] <- estimateDiversity(tse, index = "inverse_simpson", assay.type = assay_type)$inverse_simpson
+  
+  return(tse)
+}
+```
+
+2. Let's also define a function for extracting the diversity data frame from a `tse` object, and melting it into long format, for easy plotting with `ggplot2`.
+
+```{r}
+create_diversity_df <- function(tse, assay_type) {
+  # Cretate the data frame
+  diversity_df <- data.frame(
+    Sample = rownames(colData(tse)),
+    Observed = colData(tse)[[paste0("observed_", assay_type)]],
+    Chao1 = colData(tse)[[paste0("chao1_", assay_type)]],
+    ACE = colData(tse)[[paste0("ace_", assay_type)]],
+    Shannon = colData(tse)[[paste0("shannon_", assay_type)]],
+    Simpson = colData(tse)[[paste0("simpson_", assay_type)]]
+  )
+  
+  # Melt the data frame
+  melted_df <- melt(diversity_df, id.vars = c("Sample"), measure.vars = c("Observed", "Chao1", "ACE", "Shannon", "Simpson"), 
+                    variable.name = "measure", value.name = "value")
+
+  return(melted_df)
+}
+```
+
+3. Now, prepare our melted data frames on our non-normalized data, our css normalized data, and our tss normalized data.
+
+```{r}
+# Prepare df with no normalization
+normal_tse <- calculate_diversity_measures(filtered_tse, "counts")
+normal_melted_df <- create_diversity_df(normal_tse, "counts")
+
+# Prepare our rarefied df
+rarefied_tse <- calculate_diversity_measures(filtered_tse, "rarefy")
+rarefied_melted_df <- create_diversity_df(rarefied_tse, "rarefy")
+
+# Round the css data for this example - keeping the original css data for later
+rounded_tse <- filtered_tse
+assay(rounded_tse, "css") <- round(assay(rounded_tse, "css"))
+
+# Prepare our css df
+css_tse <- calculate_diversity_measures(rounded_tse, "css")
+css_melted_df <- create_diversity_df(css_tse, "css")
```

Comment:
These all should be updated to reflect the suggested changes from the above reviews